### ISSUE TITLE
Experiencing IC issues with `com.squareup.anvil.trackSourceFiles=true` in embedded mode

### ISSUE THREAD

**Issue by r0adkll** [square/anvil#980]
After updating our project to `2.5.0-beta0*` (_we're experiencing this issue in `beta03` through `beta07`_) we are still experiencing the IC issue where generated anvil code is deleted and not restored during incremental compilation unless the change itself was for a file our custom code generator picks up. We've migrated all our custom code generators to support and use `GeneratedFileWithSources` but this issue still seems present with any IC change for a given module. 

I can see the `generated-file-cache.bin` file for the given module and it looks to contain all the necessary generated class files but does not seem to be restoring them during compilation after the anvil output folder is wiped and I'm not exactly sure what is going on.

--- Edit

Okay I think I have a solid repro case now. Here is my sample repo to reproduce the bug: https://github.com/r0adkll/anvil-ic-bug-repro

The kicker I believe is this line:

```kotlin
allprojects {
  layout.buildDirectory.set(File("${rootProject.rootDir}/moduleBuildDirs/${rootProject.relativePath(project.projectDir)}/build"))
}
```

In our project repo we set a common build dir output just as above for all "projects" in the repository. After doing so in my above sample you could start to reproduce the issue we are seeing with the follow steps:

## Steps to reproduce
1. Clean build the project, and check that all the expected Anvil output is there (i.e. `moduleBuildDirs/feed/impl/build/anvil/generated/...`)
2. Make _any_ change inside the `:feeds:impl` module (_I tested with adding changes to `FeedRepositoryImpl` just so you can see the partial Anvil generated output_)
3. Incrementally compile the `:feeds:impl` module
4. **Notice that the expected Anvil generated output is no longer there _(unless you modified an Anvil annotated class)_**

If you comment out the build dir line above in the `build.gradle.kts` file and repeat the steps above you'll notice that the expected Anvil output does not get cleared on IC builds

**Comment by r0adkll** [comment]
Just updating the ticket here after talking with @RBusarow and it looks like his PR here, https://github.com/square/anvil/pull/979, fixes this issue in our project

**Comment by r0adkll** [comment]
Closing this as fixed with #979
